### PR TITLE
[angle] Improve port

### PR DESCRIPTION
**Pull request by Neumann-A**
- update opengl-registry
- give opengl-registry headers priority
- allow static builds 
- make cmake build more equivalent to native builds.

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@Neumann-A, glad failed on windows all triplets on CI testing, does this expect?

**Comment by Neumann-A** [comment]
@PhoebeHui It seems to be a file encoding problem. After changing the encoding of the *.xml files from UTF-8 with BOM to only UTF-8 `glad` compiled fine.  The problem is I don't know the best way to remove the BOM with CMake.

File: ports/angle/portfile.cmake [hunk 1]
```diff
@@ -1,4 +1,4 @@
-vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+#vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
```
Comment (JackBoosY) [inline ports/angle/portfile.cmake]: Can you just remove this line?

File: ports/angle/CMakeLists.txt [hunk 2]
```diff
@@ -361,20 +390,62 @@ add_library(libEGL
...
+target_include_directories(libEGL PUBLIC "$<INSTALL_INTERFACE:include>" "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>")
+target_compile_definitions(libEGL
+    PRIVATE -DLIBGLESV2_IMPLEMENTATION
+    PUBLIC
+        "-DGL_GLES_PROTOTYPES=1"
+        "-DGL_GLEXT_PROTOTYPES"
+        "-DEGL_EGL_PROTOTYPES=1"
+        "-DEGL_EGLEXT_PROTOTYPES"
+)
+if(BUILD_SHARED_LIBS)
```
Comment (JackBoosY) [inline ports/angle/CMakeLists.txt]: Since the port already supports dynamic build, should we also define the `GL_API` as `__declspec (dllexport)`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/angle/CMakeLists.txt (hunk 2 above)
Comment (Neumann-A) [reply ports/angle/CMakeLists.txt]: angle comes with *.def files so this is not required. 
For consumers `__declspec (dllimport)` will be defined by `KHRONOS_APICALL`

Comment (JackBoosY) [reply ports/angle/CMakeLists.txt]: Okay, that's make sence.

**Review by JackBoosY — APPROVED** [review]

**Comment by dan-shaw** [comment]
@Neumann-A Sorry for the delay. During the weekend, I will review your PRs: #9965, #9864, #9861, #9705 and more.

**Review by dan-shaw — APPROVED** [review]

File: ports/angle/CMakeLists.txt [hunk 3]
```diff
@@ -361,19 +390,55 @@ add_library(libEGL
...
+        )
+    endif()
+endif()
 
 SET_TARGET_PROPERTIES(libANGLE PROPERTIES PREFIX "")
 SET_TARGET_PROPERTIES(libGLESv2 PROPERTIES PREFIX "")
 SET_TARGET_PROPERTIES(libEGL PROPERTIES PREFIX "")
 
-install(TARGETS libEGL libGLESv2 EXPORT ANGLEExport
+install(TARGETS libEGL libGLESv2 libANGLE angle_common angle_image_util angle_gpu_info_util angle_translator angle_preprocessor angle_renderer_opengl EXPORT ANGLEExport
```
Comment (vserdyuk) [inline ports/angle/CMakeLists.txt]: Even when using x64-windows triplet which uses dynamic linkage, libANGLE.lib is put into installed\x64-windows\lib
and it's about 830 MiB in size. Together with installed\x64-windows\debug\lib\libANGLE.lib (880 MiB) this single line adds 1.67 GiB of unnecessary bloat. Please, install static libraries only when x64-windows-static is used.

**Pull request merged**